src/main/java/com/exasol/ReleaseRobot.java

```diff
@@ -1,88 +1,171 @@
+    private final String repositoryOwner;
+
+    private ReleaseRobot(final Builder builder) {
+        this.gitBranch = builder.gitBranch;
+        this.goal = builder.goal;
+        this.platforms = builder.platforms;
+        this.repositoryName = builder.repositoryName;
+        this.repositoryOwner = builder.repositoryOwner;
+    }
 
     /**
      * Main entry point for all Release Robot's calls.
-     * 
-     * @param repositoryName name of a target project from GitHub
-     * @param goalAsString run goal. Supported goals: release, validate
-     * @param platforms one or more platforms for validation or release. Supported values: github
      */
-    public void dispatch(final String repositoryName, final String goalAsString, final String... platforms) {
-        LOGGER.fine(MessageFormat.format("Release Robot has received '{}' request for the project '{}'.", goalAsString,
-                repositoryName));
+    public void run() {
+        LOGGER.fine(() -> "Release Robot has received '" + this.goal + "' request for the project "
+                + this.repositoryName + ".");
         try {
-            final Goal goal = Goal.getGoal(goalAsString);
-            final RepositoryHandler repositoryHandler = getRepositoryHandler(repositoryName, platforms);
-            if (goal == Goal.VALIDATE) {
-                repositoryHandler.validate();
+            final GitRepository repository = GitRepositoryFactory.getInstance()
+                    .createGitHubGitRepository(this.repositoryOwner, this.repositoryName);
+            final RepositoryHandler repositoryHandler = new RepositoryHandler(repository, this.platforms);
+            if (this.goal == Goal.VALIDATE) {
+                runValidation(repositoryHandler);
             } else {
-                repositoryHandler.validate();
-                repositoryHandler.release();
+                runRelease(repositoryHandler);
             }
         } catch (final RuntimeException exception) {
-            LOGGER.severe(MessageFormat.format("'{}' request failed. Cause: {}", goalAsString, exception.getMessage()));
+            LOGGER.severe(() -> "'" + this.goal + "' request failed. Cause: " + exception.getMessage());
         }
     }
 
-    private RepositoryHandler getRepositoryHandler(final String repositoryName, final String[] platforms) {
-        final Set<ReleasePlatform> platformsList = ReleasePlatform.toSet(platforms);
-        final CredentialsProvider credentialsProvider = CredentialsProvider.getInstance();
-        final GitHubRepository repository = GitHubRepositoryFactory.getInstance().createGitHubRepository(
-                REPOSITORY_OWNER, repositoryName, credentialsProvider.provideGitHubCredentials());
-        return new RepositoryHandler(repository, platformsList);
+    private void runValidation(final RepositoryHandler repositoryHandler) {
+        if (this.gitBranch == null || this.gitBranch.isEmpty()) {
+            repositoryHandler.validate(Optional.empty());
```

Comment:
Never use `Optional` as parameter. Overload the function instead.
`Optional`s intended use is as return parameter in cases where you can't be sure you really get a result.
Using it as a parameter forces switch logic in the method (which usually violates the SRP for that method). Also `null` checking gets harder and performance is lost.